src/main/java/org/gbif/doi/service/datacite/RestJsonApiDataCiteService.java

```diff
@@ -0,0 +1,206 @@
+        Preconditions.checkNotNull(doi);
+        Preconditions.checkNotNull(metadata);
+        DoiSimplifiedModel model = prepareDoiCreateModel(doi, metadata);
+        JSONAPIDocument<DoiSimplifiedModel> jsonApiWrapper = new JSONAPIDocument<>(model);
+        dataCiteClient.createDoi(jsonApiWrapper);
+    }
+
+    /**
+     * Reserve a doi. Uses createDoi method without the event type.
+     * @param doi the identifier
+     * @param metadata the metadata to be associated with the doi
+     */
+    @Override
+    public void reserve(DOI doi, DataCiteMetadata metadata) throws DoiException {
+        Preconditions.checkNotNull(doi);
+        Preconditions.checkNotNull(metadata);
+        String xmlMetadata = DataCiteValidator.toXml(doi, metadata);
+        reserve(doi, xmlMetadata);
+    }
+
+    /**
+     * Register a doi. Uses createDoi method with the event type 'PUBLISH'.
+     * @param doi the identifier
+     * @param metadata the metadata to be associated with the doi
+     */
+    @Override
+    public void register(DOI doi, URI target, String metadata) {
+        Preconditions.checkNotNull(doi);
+        Preconditions.checkNotNull(metadata);
+        DoiSimplifiedModel model = prepareDoiCreateModel(doi, metadata);
+        model.setEvent(EventType.PUBLISH.getValue());
+        JSONAPIDocument<DoiSimplifiedModel> jsonApiWrapper = new JSONAPIDocument<>(model);
+        dataCiteClient.createDoi(jsonApiWrapper);
+    }
+
+    /**
+     * Register a doi. Uses createDoi method with the event type 'PUBLISH'.
+     * @param doi the identifier
+     * @param metadata the metadata to be associated with the doi
+     */
+    @Override
+    public void register(DOI doi, URI target, DataCiteMetadata metadata) throws DoiException {
+        Preconditions.checkNotNull(doi);
+        Preconditions.checkNotNull(metadata);
+        String xmlMetadata = DataCiteValidator.toXml(doi, metadata);
+        register(doi, target, xmlMetadata);
+    }
+
+    /**
+     * Creates DoiSimplifiedModel which can be passed as an argument to dataCiteClient's create method.
+     * @param doi the identifier
+     * @param metadata the metadata to be associated with the doi
+     * @return doi model which can be registered or reserved
+     */
+    private DoiSimplifiedModel prepareDoiCreateModel(DOI doi, String metadata) {
+        DoiSimplifiedModel model = new DoiSimplifiedModel();
+        model.setId(doi.getDoiName());
+        model.setDoi(doi.getDoiName());
+        model.setXml(Base64.getEncoder().encodeToString(metadata.getBytes()));
+        return model;
+    }
+
+    /**
+     * Delete a doi.
+     * @param doi the identifier to delete
+     * @return true if successfully deleted, false otherwise
+     */
+    @Override
+    public boolean delete(DOI doi) {
+        Preconditions.checkNotNull(doi);
+        Response<Void> deleteResponse = dataCiteClient.deleteDoi(doi.getDoiName());
+        return deleteResponse.code() == 204;
```

Comment:
is there any reason to not use deleteResponse.isSuccessful()?

Reply:
fixed